src/shared/libcrypt-util.c

```diff
@@ -90,6 +90,19 @@ int make_salt(char **ret) {
 #endif
 }
 
+#if !HAVE_CRYPT_RA
+/* Provide a poor man's fallback that uses a fixed size buffer. */
+
+static char* systemd_crypt_ra(const char *phrase, const char *setting, void **data, int *size) {
+        struct crypt_data cd = {};
```

Comment:
i think we should malloc() this, given that the structure might be too large for the stack according to the man page:

"struct crypt_data may be quite large (32kB in this implementation of libcrypt; over 128kB in some other implementations).  This is large enough that it may be unwise to allocate it on the stack".